src/Peer.ts

```diff
@@ -177,7 +178,19 @@ export class Peer {
         }
       }, PEER_CONSTANTS.EXPIRATION_LOOP_INTERVAL)
 
+    const scheduleUpdateNetwork = () =>
+      setTimeout(() => {
+        try {
+          this.updateNetwork()
```

Comment:
I'd replace this with `triggerUpdateNetwork`, which puts a "name" to each network update.

If not, keep in mind that `updateNetwork` is async, so you may want to use `.catch()` instead of `try/catch`, or async/await

Reply:
updated